server/routes/viewModels/taskLists/caTasks.js

```diff
@@ -116,21 +118,21 @@ module.exports = {
     const proposedAddressTask = {
       title: 'Proposed curfew address',
       label: curfewAddress.getLabel({ decisions, tasks }),
-      action: curfewAddress.getCaProcessingAction({ decisions, tasks }),
+      action: curfewAddress.getCaProcessingAction({ decisions, tasks }, 'proposed-curfew-address'),
       visible: !bassReferralNeeded && allowedTransition !== 'caToRo',
     }
 
     const curfewAddressTask = {
       title: 'Curfew address',
       label: proposedAddress.getLabel({ decisions, tasks }),
-      action: proposedAddress.getCaAction({ decisions, tasks }),
+      action: proposedAddress.getCaAction({ decisions, tasks }, 'curfew-address'),
```

Comment:
would have been tempted to change `getCaAction()` so that this was returned from that level rather than pushed into here. 

It's about encapsulation - this code was delegating to the proposedAddress to retrieve information about this button - this change means that we're now telling it some of the information about what we want it to return!

Reply:
done for proposedAddress and the other getCaAction functions too